internal/cmd/test.go

```diff
@@ -138,3 +139,20 @@ func runCMD(mainFunc func(), args []string) (err error) {
 
 	return err
 }
+
+func RunCommonCMDTests(test *testing.T, mainFunc func(), args []string, expectedExitCode int, index int) (output string, errs error) {
```

Comment:
I don't like the index here (since not all tests uses the test cases pattern).
Instead, pass a prefix (string) that the caller can build.
Tests with test cases can build it (e.g. `fmt.Sprintf("Case %d: ", i)`),
and other tests can just pass an empty string.